### PR TITLE
Rxjava2: Initial migration to RxJava2

### PR DESCRIPTION
**Pull request by UeliKurmann**
This is an initial migration to RxJava2 and could be a base for a RxSwing2 stream.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 }
 
 apply plugin: 'nebula.rxjava-project'
 apply plugin: 'java'
+sourceCompatibility = JavaVersion.VERSION_1_8
+targetCompatibility = JavaVersion.VERSION_1_8
 
 dependencies {
-    compile 'io.reactivex:rxjava:1.1.8'
+    compile 'io.reactivex.rxjava2:rxjava:2.0.0'
```
Comment (akarnokd) [inline build.gradle:14]: 2.0.1

**Review by akarnokd — COMMENTED** [review]
Maybe post the whole to RxSwing2?

**Comment by UeliKurmann** [comment]
The travis build failed because JDK 1.7 is used. Is this configured in travis?

File: build.gradle (hunk 1 above)
Comment (UeliKurmann) [reply build.gradle:14]: OK, i will try to post it to RxSwing2 and revoke this pull request.

**Comment by akarnokd** [comment]
You have to change `.travis.yml`

```
jdk:
  - oraclejdk8
# force upgrade Java8 as per https://github.com/travis-ci/travis-ci/issues/4042 (fixes compilation issue)
addons:
  apt:
    packages:
      - oracle-java8-installer
```

**Pull request closed without merging**